WoWsShipBuilder.Core/DataContainers/AccelerationHelper.cs

```diff
@@ -0,0 +1,416 @@
+    /// <term>Advanced info</term>
+    /// <description> https://bbs.nga.cn/read.php?tid=27647748</description>
+    /// </item>
+    /// <item>
+    /// <term>Speed boost info</term>
+    /// <description> https://bbs.nga.cn/read.php?tid=31558840</description>
+    /// </item>
+    /// </list>
+    /// </summary>
+    /// <param name="shipIndex">Index of the ship.</param>
+    /// <param name="hull">Hull module of the ship.</param>
+    /// <param name="engine">Engine module of the ship.</param>
+    /// <param name="shipClass">Class of the ship.</param>
+    /// <param name="throttleList">List of throttle to use. First is the starting throttle.</param>
+    /// <param name="speedMultiplier">Multiplier for the base speed to apply.</param>
+    /// <param name="engineForwardUpTimeModifiers">Multipliers for the engine forward up time to apply.</param>
+    /// <param name="engineBackwardUpTimeModifiers">Multipliers for the engine backwards up time to apply.</param>
+    /// <param name="engineForwardForsageMaxSpeedModifier">Multipliers for the forward forsage max speed to apply. <b>IMPORTANT:</b> speed boost <b>OVERRIDE</b> this parameter, it does not stack.</param>
+    /// <param name="engineBackwardForsageMaxSpeedModifier">Multipliers the backward forsage max speed to  to apply. <b>IMPORTANT:</b> speed boost <b>OVERRIDE</b> this parameter, it does not stack.</param>
+    /// <param name="engineForwardForsagePowerModifier">Multipliers for the forward forsage power to apply. <b>IMPORTANT:</b> for speed boost <b>USE</b> the speedBoostEngineForwardForsageMaxSpeedOverride parameter. Speed boost is not a multiplier, but override the value directly.</param>
+    /// <param name="engineBackwardForsagePowerModifier">Multipliers for the backward forsage power to apply. <b>IMPORTANT:</b> for speed boost <b>USE</b> the speedBoostEngineForwardForsageMaxSpeedOverride parameter. Speed boost is not a multiplier, but override the value directly.</param>
+    /// <param name="speedBoostEngineForwardForsageMaxSpeedOverride">Value from speed boost that override the actual engineForwardForsageMaxSpeed. Set to 0 if not present.</param>
+    /// <param name="speedBoostEngineBackwardEngineForsagOverride">Value from speed boost that override the actual engineBackwardForsageMaxSpeed. Set to 0 if not present.</param>
+    /// <param name="forwardEngineForsagOverride">Value from speed boost that override the actual forwardEngineForsag. Set to 0 if not present.</param>
+    /// <param name="backwardEngineForsagOverride">Value from speed boost that override the actual backwardEngineForsag. Set to 0 if not present.</param>
+    /// <exception cref="InvalidOperationException">Thrown when the throttleList contains invalid numbers.</exception>
+    /// <exception cref="OverflowException">Thrown when the calculation takes too many iterations.</exception>
+    /// <returns>The data regarding acceleration.</returns>
+    public static AccelerationData CalculateAcceleration(
```

Comment:
Refactor to pass less parameters, group them into records or even record structs if necessary

Reply:
For this and the similar ones: i think the addition of a record/struct just to pass parameters to a method would just obscure/make the code harder to read. having all the parameters visible at once is easier to read. Having a record would just obsfuscate it more, while also just splitting the call on multiple lines, with no real advantage (and also having to "unpack" them on the method itself

Reply:
While that is not wrong, the current state is not an acceptable solution either because the amount of parameters makes it very difficult to read.

Reply:
Done